core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -1148,6 +1206,22 @@ public boolean isNeededDependenciesForNewerJenkins(PluginManager.MetadataCache c
             });
         }
 
+        /**
+         * Returns true iff this plugin or any of its dependencies require a newer Java than Jenkins is running on.
+         *
+         * @return true iff this plugin or any of its dependencies require a newer Java than Jenkins is running on.
```

Comment:
The comment doesn't fit the code, I think. The code is not taking into account if this plugin *isForNewerJava*, it only checks its dependencies. Or I'm missing something. I don't know whether the code is wrong, or just the comment

Reply:
fixed, nice catch.